internal/v2/s2av2.go

```diff
@@ -131,7 +132,22 @@ func (c *s2av2TransportCreds) ClientHandshake(ctx context.Context, serverAuthori
 	serverName := removeServerNamePort(serverAuthority)
 	timeoutCtx, cancel := context.WithTimeout(ctx, GetS2ATimeout())
 	defer cancel()
-	s2AStream, err := createStream(timeoutCtx, c.s2av2Address, c.getS2AStream)
+	var err error
+	var s2AStream stream.S2AStream
+	retryer := NewRetryer()
+	for {
```

Comment:
We have this same codeblock a lot. WDYT of having helper function that does this instead? It could accept like `func()` to keep it generic.

Reply:
updated added a helper function in retry.go
PTAL
`func Run(ctx context.Context, f func() error) {`